src/internal/connector/onedrive/collections.go

```diff
@@ -180,7 +180,15 @@ func deserializeMetadata(
 
 		// Go through and remove partial results (i.e. path mapping but no delta URL
 		// or vice-versa).
-		for k := range prevDeltas {
+		for k, v := range prevDeltas {
+			// Remove entries with an empty delta token as it's not useful.
+			if len(v) == 0 {
```

Comment:
Just wondering what is the scenario where this would happen.

Reply:
I don't expect this to happen at the moment, but I do want to make sure we handle it if it does. Right now the code that serializes the delta tokens has it's own check to ensure no 0 length tokens are added to the metadata. However, having a check here makes the code a little more flexible because even if the other check is accidentally removed things should (hopefully) continue working as expected